testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -601,6 +604,119 @@ public void testExplainAnalyze()
         assertExplainAnalyze("EXPLAIN ANALYZE SHOW SESSION");
     }
 
+    @Test
+    public void testMaterializedViewsMetadata()
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW)) {
+            return;
+        }
+
+        @Language("SQL") String materializedViewsResultRow = format(
+                "VALUES ('%s', 'tpch', '%s', 'true', 'user', 'sarcastic comment', '%s')",
+                getCatalogName(),
+                MATERIALIZED_VIEW_NAME,
+                materializedViewDefinitionSql());
+
+        computeActual(format(
+                "CREATE MATERIALIZED VIEW %s COMMENT 'sarcastic comment' AS %s",
+                MATERIALIZED_VIEW_NAME,
+                materializedViewDefinitionSql()));
+
+        computeActual(format(
+                "CREATE MATERIALIZED VIEW %s COMMENT 'sarcastic comment' AS %s",
+                OTHER_MATERIALIZED_VIEW_NAME,
+                materializedViewDefinitionSql()));
+
+        // test freshness
+        assertQuery(
+                selectMaterializedViewsSql(),
+                getMaterializedViewsResultRows(false));
+
+        computeActual(format("REFRESH MATERIALIZED VIEW %s", MATERIALIZED_VIEW_NAME));
+
+        assertQuery(
+                selectMaterializedViewsSql(),
+                getMaterializedViewsResultRows(true));
+
+        // test filtering by columns
+        assertQuery(
+                selectMaterializedViewsSql(format("catalog_name = '%s'", getCatalogName())),
+                getMaterializedViewsResultRows(true));
+
+        assertQuery(
+                selectMaterializedViewsSql(
+                        format("catalog_name = '%s'", getCatalogName()),
+                        format("schema_name = '%s'", getSchemaName())),
+                getMaterializedViewsResultRows(true));
```

Comment:
Please put `OTHER_MATERIALIZED_VIEW_NAME` in another schema, you can create a schema in this method for testing.

Reply:
That's a good point